### PR TITLE
Defer write

### PR DESCRIPTION
**Pull request by greg-1-anderson**
Defer execution of operations in taskWriteToFile until the run() method, in case a file referenced in textFromFile() might not be created yet.

This is more important with #252.  Note that I have once again merged in #251 here, so that unit tests will pass.  This should not cause any merge conflicts.

### CODE REVIEW

**Comment by greg-1-anderson** [comment]
Good suggestions. I updated the PR to address.

File: src/Task/File/Write.php [hunk 1]
```diff
@@ -121,20 +126,49 @@ public function replace($string, $replacement)
...
-        $this->body = preg_replace($pattern, $replacement, $this->body);
+        $this->stack[] = array_merge(['_' . __FUNCTION__], func_get_args());
         return $this;
     }
 
+    protected function _textFromFile($contents, $filename) {
+        return $contents . file_get_contents($filename);
+    }
+
+    protected function _replace($contents, $string, $replacement) {
```
Comment (DavertMik) [inline src/Task/File/Write.php]: protected functions shound not start with `_`. It violates PSR standard: https://github.com/php-fig/fig-standards/blob/master/accepted%2FPSR-2-coding-style-guide.md#43-methods

**Comment by greg-1-anderson** [comment]
Renamed `_foo()` to `fooCollect`

**Comment by DavertMik** [comment]
I'm ok with it. @burzum If you have no objections, please merge it

**Pull request merged**